modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponentTestCase.java

```diff
@@ -0,0 +1,141 @@
+
+  @Rule
+  public MockitoRule rule = MockitoJUnit.rule();
+
+  @Mock
+  private FeatureFlaggingService featureFlaggingService;
+
+  @Mock
+  private MuleApplicationClassLoader applicationClassLoader;
+
+  private ExtensionComponent extensionComponent;
+
+  private AtomicReference<ClassLoader> executedClassloader = new AtomicReference<>();
+
+  private MuleArtifactClassLoader artifactClassLoader;
+
+  @Before
+  public void setUp() throws Exception {
+    extensionComponent = new TestExtensionComponent(mock(ExtensionModel.class),
+                                                    mock(ComponentModel.class), mock(ConfigurationProvider.class),
+                                                    mock(CursorProviderFactory.class), mock(ExtensionManager.class));
+
+    extensionComponent.featureFlaggingService = featureFlaggingService;
+
+    RegionClassLoader regionClassLoader = mock(RegionClassLoader.class);
+    when(regionClassLoader.getOwnerClassLoader()).thenReturn(applicationClassLoader);
+    when(applicationClassLoader.getClassLoader()).thenReturn(applicationClassLoader);
+
+    URL url = new URL("file:///app.txt");
+    artifactClassLoader = new MuleArtifactClassLoader("test", mock(ArtifactDescriptor.class),
+                                                      new URL[] {url}, regionClassLoader,
+                                                      mock(ClassLoaderLookupPolicy.class));
+
+    extensionComponent.classLoader = artifactClassLoader;
+
+    MetadataCacheIdGeneratorFactory metadataCacheIdGeneratorFactory = mock(MetadataCacheIdGeneratorFactory.class);
+    when(metadataCacheIdGeneratorFactory.create(any(), any())).thenReturn(null);
+    extensionComponent.setCacheIdGeneratorFactory(metadataCacheIdGeneratorFactory);
+    executedClassloader.set(null);
+  }
+
+
+
+  @Test
+  public void testLoadingWithComposite() throws MuleException {
+    when(featureFlaggingService.isEnabled(START_EXTENSION_COMPONENTS_WITH_ARTIFACT_CLASSLOADER)).thenReturn(false);
+    extensionComponent.initialise();
+    extensionComponent.start();
+    assertThat(executedClassloader.get(), instanceOf(CompositeClassLoader.class));
+    assertThat(((CompositeClassLoader) executedClassloader.get()).getDelegates().size(), is(2));
+    assertThat(((CompositeClassLoader) executedClassloader.get()).getDelegates().get(0), is(artifactClassLoader));
+    assertThat(((CompositeClassLoader) executedClassloader.get()).getDelegates().get(1), is(applicationClassLoader));
+  }
+
+  @Test
+  public void testLoadingWithArtifact() throws MuleException {
+    when(featureFlaggingService.isEnabled(START_EXTENSION_COMPONENTS_WITH_ARTIFACT_CLASSLOADER)).thenReturn(true);
+    extensionComponent.initialise();
+    extensionComponent.start();
+    assertThat(executedClassloader.get(), instanceOf(MuleArtifactClassLoader.class));
+    assertThat(executedClassloader.get(), is(artifactClassLoader));
+  }
+
+
+  private final class TestExtensionComponent extends ExtensionComponent {
```

Comment:
why nor mock this instead of exteniding it for a test class?

Reply:
To get the executedClassloader when the doStart method is called inside the withContextClassloader(classloader)